event.py

```diff
@@ -0,0 +1,91 @@
+from typing import Iterable
+from copy import deepcopy
+from flask_restful import abort
+
+from timemodel import Time
+from eventdao import EventDAO
+
+
+DAYS = set(('Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'))
+
+class Event:
+    dao = EventDAO()
+
+    def __init__(self, name: str, days: Iterable[str], from_time: Time, to_time: Time, description=''):
+        assert set(days).issubset(DAYS)
+        assert from_time.in_seconds() <= to_time.in_seconds()
+
+        self.name = name
+        self.days = set(days)
+        self.from_time = from_time
+        self.to_time = to_time
+        self.description = description
+
+    @classmethod
+    def fromdict(cls, event_dict):
+        name = event_dict['name']
+        days = event_dict['days']
+        from_time = Time.fromdict(event_dict['from_time'])
+        to_time = Time.fromdict(event_dict['to_time'])
+        description = event_dict['description'] if 'description' in event_dict else ''
+        return cls(name, days, from_time, to_time, description)
+
+    def serialize(self):
```

Comment:
No need for this method, as well, all the assignments will throw a KeyError exception if the key does not exist within the event_dict

Reply:
Just going to summarize what we discussed here:
I could just use `Event.__dict__` to get a dictionary of the members and their values, but in the case that the values of an Object also contain Objects (so entities that are not JSON serializable), I have to call `__dict__` on those as well recursively until all values are JSON serializble (so either strings, numbers , lists or dicts). In this case, I can remove my own `Time` model class since it has been done before, but `Schedule` holds a list of Event objects, so it will have to have its own serialize function, and for consistency, I'll keep one for this model class as well.